tnc/data_handler.py

```diff
@@ -1830,20 +1914,25 @@ def burst_watchdog(self):
         DATA BURST
         """
         # IRS SIDE
+        # TODO: We need to redesign this part for cleaner state handling
         if not static.ARQ_STATE or static.ARQ_SESSION_STATE != 'connected' or static.TNC_STATE != 'BUSY' or not self.is_IRS:
-            return
-
+            # return only if not ARQ STATE and not ARQ SESSION STATE as they are different use cases
+            if not static.ARQ_STATE and static.ARQ_SESSION_STATE == 'disconnected' or not self.is_IRS:
+                return
```

Comment:
Yes, lets, try `sourcery`'s suggestion, as I also agree. However, we need to have a sharp eye on these tnc states. I'm not happy with the existing solution how this will be handled tnc wide 🤔